custom_enumerable_methods.rb

```diff
@@ -0,0 +1,113 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self # rubocop:disable Style/For
+      yield(element)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = self
+    i = 0
+    element = 0
+    my_self.my_each do
+      yield(element, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_self = self
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?(*arg)
+    result = true
+    if !arg[0].nil?
+      my_each { |i| result = false unless arg[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
+
+  def my_any?(*arg)
+    result = false
+    if !arg[0].nil?
+      my_each { |i| result = true if arg[0] == i }
+    elsif !block_given?
+      my_each { |item| result = true if item }
+    else
+      my_each { |item| result = true if yield(item) }
+    end
+    result
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
```

Comment:
```rb
2.6.5 :054 > array
 => [3, 4, 2, 6, 8, 4, 6, 5, 1, 2, 4, 1, 5, 6, 7, 8, 1, 7, 3, 0, 8, 0, 3, 6, 0, 1, 0, 6, 2, 6, 3, 7, 3, 2, 3, 5, 1, 1, 7, 8, 2, 3, 4, 8, 1, 8, 0, 3, 4, 5, 4, 3, 3, 7, 1, 8, 4, 7, 8, 2, 6, 8, 0, 6, 0, 5, 1, 2, 5, 2, 2, 4, 0, 2, 1, 8, 1, 3, 3, 8, 2, 8, 1, 6, 1, 0, 1, 2, 5, 0, 5, 8, 1, 3, 2, 7, 4, 5, 2, 4] 
2.6.5 :055 > array.my_none?(String)
 => true 
2.6.5 :056 > array.none?(String)
 => true 
2.6.5 :057 > arr = array.dup
2.6.5 :058 > arr[0] = 'hi'
2.6.5 :059 > array.none?(String)
 => true 
2.6.5 :060 > array.my_none?(String)
 => true 
2.6.5 :061 > 
2.6.5 :062 > arr.my_none?(String)
 => true 
2.6.5 :063 > arr.none?(String)
 => false 
2.6.5 :064 >
```

Reply:
```rb
2.6.5 :066 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :067 > words.my_none?(/z/)
 => true 
2.6.5 :068 > words.none?(/z/)
 => true 
2.6.5 :069 > words.my_none?(/d/)
 => true 
2.6.5 :070 > words.none?(/d/)
 => false 
2.6.5 :071 >
```

Reply:
It seems to be working well.

p array.my_none?(String)
p array.none?(String)
 arr = array.dup
arr[0] = 'hi'
p array.none?(String)

p array.my_none?(String)

p arr.my_none?(String)
 
p arr.none?(String)

#true
#true
#true
#true
#false
#false


 words =["dog", "door", "rod", "blade"] 
p words.my_none?(/z/)
p words.none?(/z/)
p words.my_none?(/d/)
p words.none?(/d/)

#true
#true
#false
#false

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more
### Thank You so much! :bouquet: